apiserver/facades/agent/uniter/networkinfocaas.go

```diff
@@ -165,23 +169,36 @@ func (n *NetworkInfoCAAS) NetworksForRelation(
 	var ingress corenetwork.SpaceAddresses
 	var err error
 
+	if err = n.setCrossModelStatus(rel); err != nil {
+		return "", nil, nil, errors.Trace(err)
+	}
+
 	if pollAddr {
-		if ingress, err = n.maybeGetUnitAddress(rel); err != nil {
+		if ingress, err = n.maybeGetUnitAddress(rel, false); err != nil {
 			return "", nil, nil, errors.Trace(err)
 		}
 	}
 
+	// Ingress addresses can only be public addresses for CAAS.
+	// They are always scoped explicitly. See: caas/kubernetes/provider/k8s.go.
```

Comment:
Except that for clusterIP services, the scope is set to cloud local. We still need to include cluster IP addresses if that's how the service has been configured.
```
	case core.ServiceTypeClusterIP:
		appendUniqueAddrs(network.ScopeCloudLocal, clusterIP)
```